src/networkd.c

```diff
@@ -32,6 +33,68 @@
 #include "util-internal.h"
 #include "validation.h"
 
+/**
+ * Query sysfs for the MAC address (up to 20 bytes for infiniband) of @ifname
+ * The caller owns the returned string and needs to free it.
+ */
+STATIC char*
+_netplan_sysfs_get_mac_by_ifname(const char* ifname, const char* rootdir)
+{
+    g_autofree gchar* content = NULL;
+    g_autofree gchar* sysfs_path = NULL;
+    sysfs_path = g_build_path(G_DIR_SEPARATOR_S, rootdir ?: G_DIR_SEPARATOR_S,
+                              "sys", "class", "net", ifname, "address", NULL);
+
+    if (!g_file_get_contents (sysfs_path, &content, NULL, NULL)) {
+        g_debug("%s: Cannot read file contents.", __FUNCTION__);
+        return NULL;
+    }
+
+    // Trim whitespace & clone value
+    return g_strstrip(content);
+}
+
+/**
+ * Query sysfs for the driver used by @ifname
+ * The caller owns the returned string and needs to free it.
+ */
+STATIC char*
+_netplan_sysfs_get_driver_by_ifname(const char* ifname, const char* rootdir)
+{
+    g_autofree gchar* link = NULL;
+    g_autofree gchar* sysfs_path = NULL;
+    sysfs_path = g_build_path(G_DIR_SEPARATOR_S, rootdir ?: G_DIR_SEPARATOR_S,
+                              "sys", "class", "net", ifname, "driver", NULL);
+
+    link = g_file_read_link(sysfs_path, NULL);
+    if (!link) {
+        g_debug("%s: Cannot read symlink of %s.", __FUNCTION__, sysfs_path);
+        return NULL;
+    }
+
+    return g_path_get_basename(link);
+}
+
+/**
+ * Enumerate all network interfaces (/sys/clas/net/...) and check
+ * netplan_netdef_match_interface() to see if they match the current NetDef
+ */
+STATIC void
+_netplan_enumerate_interfaces(const NetplanNetDefinition* def, GStrvBuilder* builder, const char* rootdir)
```

Comment:
Is this here to work around the fact that we could have a generated file like:

```
# my-interface.network
[Match]
MacAddress=AA:BB:CC:DD:EE:FF

[Network]
...
```

but we want to enable e.g. `systemd-networkd-wait-online@ethZ.service`, if `ethZ` is the interface with the matching MAC?

Reply:
Yes! We need to expand any Netplan `match.mac|driver|name` setting to the actual matching network interfaces. Therefore, we need to check those values on all of them.